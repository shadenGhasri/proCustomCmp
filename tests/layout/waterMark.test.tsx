/* eslint-disable @typescript-eslint/ban-types */
import { WaterMark } from '@ant-design/pro-components';
import { act, render } from '@testing-library/react';

describe('WaterMark', () => {
  it('test image watermark', async () => {
    let onloadRef: Function | undefined;

    Object.defineProperty(Image.prototype, 'onload', {
      get() {
        // eslint-disable-next-line no-underscore-dangle
        return this._onload;
      },
      set(onload: Function) {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        onloadRef = onload;
        // eslint-disable-next-line no-underscore-dangle
        this._onload = onload;
      },
    });
    const { container, unmount } = render(
      <WaterMark
        rotate={0}
        image="https://img.alicdn.com/tfs/TB1YM3LpipE_u4jSZKbXXbCUVXa-280-128.png"
      >
        <div style={{ height: 500 }}>123</div>
      </WaterMark>,
    );

    act(() => {
      onloadRef?.();
    });
    expect(container).toMatchSnapshot();
    unmount();
  });

  it('test text watermark', () => {
    const wrapper = render(
      <WaterMark content="Trusple">
        <div style={{ height: 500 }} />
      </WaterMark>,
    );

    expect(wrapper.asFragment()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('test image watermark', async () => {
    const spy = jest.spyOn(global.console, 'error').mockImplementation();
    const createElement = document.createElement.bind(document);
    // @ts-ignore
    document.createElement = (tagName: string) => {
      if (tagName === 'canvas') {
        return {
          setAttribute: () => null,
          getContext: () => null,
          measureText: () => ({}),
        };
      }
      return createElement(tagName);
    };

    const { unmount } = render(
      <WaterMark
        rotate={0}
        image="https://img.alicdn.com/tfs/TB1YM3LpipE_u4jSZKbXXbCUVXa-280-128.png"
      >
        <div style={{ height: 500 }}>123</div>
      </WaterMark>,
    );

    // @ts-ignore
    expect(console.error.mock.calls).toEqual([['当前环境不支持Canvas']]);
    unmount();
    spy.mockRestore();
  });
});
