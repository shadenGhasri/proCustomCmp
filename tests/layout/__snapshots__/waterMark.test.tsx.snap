// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WaterMark test image watermark 1`] = `
<div>
  <div
    class="ant-pro-layout-watermark-wrapper"
    style="position: relative;"
  >
    <div
      style="height: 500px;"
    >
      123
    </div>
    <div
      class="ant-pro-layout-watermark"
      style="z-index: 9; position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; background-size: 332px; pointer-events: none; background-repeat: repeat; background-image: url(data:image/png;base64,00);"
    />
  </div>
</div>
`;

exports[`WaterMark test text watermark 1`] = `
<DocumentFragment>
  <div
    class="ant-pro-layout-watermark-wrapper"
    style="position: relative;"
  >
    <div
      style="height: 500px;"
    />
    <div
      class="ant-pro-layout-watermark"
      style="z-index: 9; position: absolute; left: 0px; top: 0px; width: 100%; height: 100%; background-size: 332px; pointer-events: none; background-repeat: repeat; background-image: url(data:image/png;base64,00);"
    />
  </div>
</DocumentFragment>
`;
